lib/pluginmanager/xpack/remote_installer.rb

```diff
@@ -0,0 +1,25 @@
+# encoding: utf-8
+require "pluginmanager/xpack/local_installer"
+require "pluginmanager/utils/downloader"
+require "fileutils"
+
+module LogStash module PluginManager module Xpack
+  class RemoteInstaller
+    attr_reader :remote_url, :feedback
+
+    def initialize(remote_url, feedback = Utils::Downloader::ProgressbarFeedback)
+      @remote_url = remote_url
+      @feedback = feedback
+    end
+
+    def execute
+      PluginManager.ui.info("Downloading file: #{remote_url}")
+      downloaded_file = Utils::Downloader.fetch(remote_url, feedback)
+      PluginManager.ui.debug("Downloaded package to: #{downloaded_file}")
+
+      Xpack::LocalInstaller.new(downloaded_file).execute
+    ensure
+      FileUtils.rm_rf(downloaded_file) if downloaded_file
```

Comment:
We should clear it in a rescue block there https://github.com/elastic/logstash/pull/6168/files#diff-d29ec5d7cc59e605fe75a1aace924041R66